ledger-core/v2/virtual/object/object.go

```diff
@@ -46,6 +46,8 @@ type Info struct {
 
 	AwaitPendingOrdered smachine.BargeIn
 
+	SawRequests map[reference.Global]struct{}
```

Comment:
SeenRequests/KnownRequests. Not sure about the map, we have to not only be checking if requests is in list, but also we will need to find EarliestKnownPulse. Maybe OrderedMap would suit here better (it should already be in our codebase)

Reply:
but maybe just earliest/latest request field would do the job

Reply:
@ruz do we need EarliestKnownPulse or ordered list for deduplication algorithm?

Reply:
guys, if you look at the task and the story then you can find answer to these questions.